### ISSUE TITLE
Possibility to decide which GPU is used

### ISSUE THREAD

**Issue by financelurker** [y4my4my4m/kde-shader-wallpaper#58]
Hi!

I am using these awesome wallpaper shader - cudos for the implementation.

Since I have two GPUs (one iGPU and one Nvidia) - as of now only the Nvidia GPU is used (where I have a constant 40% of GPU utilization with active wallpaper shaders).

Is it possible, that I can decide which GPU is used for these wallpaper shaders? (I would want to give that task to the iGPU - since it probably won't need that much power and I would expect a lower power consumption)

**Comment by y4my4my4m** [comment]
@financelurker 
Yes, it's possible. I don't know if wayland affects this in anyway, but I remember having tried it before on X11 and it worked.
Essentially you need to launch `plasmashell` using `DRI_PRIME=1 plasmashell --replace`  or `__NV_PRIME_RENDER_OFFLOAD=1 __GLX_VENDOR_LIBRARY_NAME=nvidia plasmashell --replace`

https://chatgpt.com/share/c17d5a99-3758-449e-af47-82a2a2c27cb3

Essentially, this isn't a plugin thing but rather you need to launch the plasmashell using your GPU instead of iGPU.
There may be a more advanced way to get the shaderengine use a specific GPU, but that would probably require modifying the wallpaper feature of KDE directly.

**Comment by financelurker** [comment]
So, to get that right - I would need to set the whole plasmashell to the iGPU and not just the Wallpaper Shaders?
I cannot gasp the impact of that, but I will try it within the next days. Thx for your feedback.

**Comment by y4my4my4m** [comment]
@financelurker 
As far as I understand it, the whole "desktop" interface of KDE runs on [Plasma-Desktop](https://invent.kde.org/plasma/plasma-desktop/-/tree/master) (Which is actually called `plasmashell`). 
This does not include all the graphical windows being displayed (that's handled by X11 and Wayland), having plasmashell run on your GPU won't make "dolphin" or "chrome" use your GPU, only the taskbar, KDE menus, etc.

In Plasmashell there is an element called "Desktop" (literally your desktop where the icons are and etc), and within that there is something called a "wallpaper", the wallpaper plugins (images/shaders/etc) are then loaded within that "wallpaper" element.

What my shader wallpaper does is limited in it's ability to go beyond the "wallpaper" element and what's available for it to use.
